src/mods.cpp

```diff
@@ -28,11 +28,44 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "util/strfnd.h"
 #include "convert_json.h"
 #include "exceptions.h"
+#include <vector>
+
+static void independentGetLine(std::istream& is, std::string& line)
+{
+    // The characters in the stream are read one-by-one using a std::streambuf.
```

Comment:
Mhh... doesn't seem like util/string.h would contain a replacement. Still, this function doesn't really belong in mods.cpp, it should be put into `util/string.{cpp,h}`.